src/solver/conv_mlir_igemm_fwd.cpp

```diff
@@ -104,8 +104,14 @@ bool PerformanceConvMlirIgemm::IsValid(const ConvolutionContext& ctx) const
 
 bool PerformanceConvMlirIgemm::SetNextValue(const ConvolutionContext& /*config*/)
 {
+    static bool allTuningExercised = false;
```

Comment:
This is not MT-safe.

Reply:
Oops, the things seem worse. Use of static variable here is totally incorrect.

Reply:
@atamazov Thanks for your review and suggestions! Let me take a closer look at the alternatives and I'll pick one of your suggestions and get it implemented.

BTW, since this will become a user-visible failure if user attempt to do tuning on their own. Do you see the urgency of this high enough to make it cherry-picked to the already merged ROCm 5.1 release?

Reply:
Done in [478fdb9](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1435/commits/478fdb99e531c3a11c0eada133b367bddc8ebcda)